crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -348,6 +394,11 @@ fn check_limits(limits: &Limits, cost: &ComponentCost, pos: Option<Pos>) -> Serv
     }
 
     if cost.output_nodes > limits.max_output_nodes {
+        if let Some(metrics) = ctx.data_opt::<Metrics>() {
+            // TODO is this a client error though?
```

Comment:
Imo yes because a caller can calculate the number of output nodes their query will produce. If it's not a client error then the onus is on us to fix

Reply:
Great, thank you!